### DISCUSSION TITLE
Cyclic workflow, is it possible?

### DISCUSSION THREAD

**Discussion by DanySK** [community/community#26391] [Actions]
<p>Hi,<br>
I’m considdering switching from Travis CI from GH Actions and I am exploring what GHA provides.<br>
I got a workflow for several repositories (dozens) that deals with automatic updates in this way:</p>
<ul>
<li>A bot runs a cronjob, fetches my repositories, fetches updates, pushes a new branch with the updates, and opens pull  requests.</li>
<li>The CI should start, check if the update works</li>
<li>Once the CI is complete, if the checks were successful, then I use the automerge action to rebase updates automatically in the development branch (the temporary branch is deleted upon rebase)</li>
</ul>
<p>In case there are updates on the development branch, the automerge action rebases the branch and force-pushes it. In this case, the CI should restart.<br>
This also captures the case in which there are concurrent updates: the first that gets merged causes a re-evaluation of further updates.</p>
<p>This workflow used to work flawlessly with Travis CI. I am having difficulties into reimplementing it with GH actions, though. It looks like that upgrades coming from force-pushes on update branches are not considered for CI. In turn, then, the automerge action does not trigger, as it is configured to start when repository checks are complete.</p>
<p>What is the right way to configure such workflow with GHA?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>I assume you’re using the <code>GITHUB_TOKEN</code> to push the updates created in the cron job? There’s a limitation to that, see <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/authentication-in-a-workflow#using-the-github_token-in-a-workflow">Using the <code>GITHUB_TOKEN</code> in a workflow</a>:</p>
<blockquote>
<p>When you use the repository’s <code>GITHUB_TOKEN</code> to perform tasks on behalf of the GitHub Actions app, events triggered by the <code>GITHUB_TOKEN</code> will not create a new workflow run. This prevents you from accidentally creating recursive workflow runs.</p>
</blockquote>
<p>To get around that limitation you’ll need to <a href="https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token">create a PAT</a> (personal access token) with the required permissions and use that instead (via <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/encrypted-secrets">Secrets</a>).</p>

**Comment by DanySK** [comment]
<p>Thank you! That was exactly the reason!</p>
